apps/api/src/app/widgets/e2e/initialize-widget-session.e2e.ts

```diff
@@ -64,48 +84,64 @@ describe('Initialize Session - /widgets/session/initialize (POST)', async () =>
     const validSecretKey = session.environment.apiKeys[0].key;
-    let hmacHash;
 
-    const invalidSubscriberId = validSubscriberId + '0';
-    hmacHash = createHash(validSecretKey, invalidSubscriberId);
+    const invalidSubscriberId = `invalid-suscriberId`;
+    const validSubscriberHmacHash = createHash(validSecretKey, subscriberId);
+    const responseInvalidSubscriberId = await initWidgetSession(invalidSubscriberId, session, validSubscriberHmacHash);
 
-    const responseInvalidSubscriberId = await initWidgetSession(validSubscriberId, session, hmacHash);
+    expect(responseInvalidSubscriberId.body?.data?.profile).to.not.exist;
+    expect(responseInvalidSubscriberId.body.message).to.contain('Please provide a valid HMAC hash');
+  });
 
-    const invalidSecretKey = validSecretKey + '0';
-    hmacHash = createHash(invalidSecretKey, validSubscriberId);
-    const responseInvalidSecretKey = await initWidgetSession(validSubscriberId, session, hmacHash);
+  it('should fail the test with invalid secret key', async function () {
+    const validSecretKey = session.environment.apiKeys[0].key;
 
-    expect(responseInvalidSubscriberId.body.message).to.contain('Please provide a valid HMAC hash');
+    const invalidSecretKey = 'invalid-secret-key';
+    const invalidSubscriberHmacHash = createHash(invalidSecretKey, subscriberId);
+
+    const responseInvalidSecretKey = await initWidgetSession(subscriberId, session, invalidSecretKey);
+
+    expect(responseInvalidSecretKey.body?.data?.profile).to.not.exist;
     expect(responseInvalidSecretKey.body.message).to.contain('Please provide a valid HMAC hash');
   });
 });
 
-async function initWidgetSession(subscriberId: string, session, hmacHash?: string) {
+async function initWidgetSession(subscriberIdentifier: string, session, hmacHash?: string) {
   return await session.testAgent.post('/v1/widgets/session/initialize').send({
     applicationIdentifier: session.environment.identifier,
-    subscriberId: subscriberId,
+    subscriberId: subscriberIdentifier,
     firstName: 'Test',
     lastName: 'User',
     email: 'test@example.com',
     phone: '054777777',
-    hmacHash: hmacHash,
+    hmacHash,
   });
 }
 
-async function setHmacConfig(session: UserSession, invalidateCache: InvalidateCacheService) {
-  const integrationRepository = new IntegrationRepository();
-
+async function setHmacConfig(userId: string, session: UserSession, invalidateCache: InvalidateCacheService) {
   await invalidateCache.invalidateQuery({
     key: buildIntegrationKey().invalidate({
       _organizationId: session.organization._id,
     }),
   });
 
-  await integrationRepository.update(
+  const command = {
+    environmentId: session.environment._id,
+    userId,
+    providerId: InAppProviderIdEnum.Novu,
+    channel: ChannelTypeEnum.IN_APP,
+  };
+
+  await invalidateCache.invalidateByKey({
+    key: buildIntegrationKey().cache({
+      _organizationId: session.organization._id,
+      ...command,
+    }),
+  });
```

Comment:
Without this cache wasn't fully erased, returning in SelectIntegration all the time the cached initial integration (the one created by UserSession) and not the updated one with the values changed from line 144.

Reply:
Not sure about that but that won't make any reader to lose knowledge of what's going on in the tests?
CC: @djabarovgeorge how is your vision on this for cache?